tests/integration/self_update_test.sh

```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+
+include './src/lib/kwio.sh'
+include './tests/unit/utils.sh'
+include './tests/integration/utils.sh'
+
+declare -g expected_output
+
+function self_update_test_helper()
+{
+  local update_type="$1"
+  local kw_branch="${update_type}"
+  local base_version
+  local distro
+  local container
+  local output
+  local expected_output
+  local head_commit_hash
+
+  # Get the short commit hash of the branch
+  head_commit_hash=$(git rev-parse --short origin/"${kw_branch}")
+
+  # Get the base version from the specified branch
+  base_version=$(git show "${kw_branch}":./src/VERSION | head --lines 1)
+
+  # Format the expected output
+  expected_output=$(printf '%s\nBranch: %s\nCommit: %s' "$base_version" "$kw_branch" "$head_commit_hash")
+
+  for distro in "${DISTROS[@]}"; do
+    say "Running tests on ${distro} environment..."
+    container="kw-${distro}"
+
+    case "$update_type" in
+      'master')
+        container_exec "${container}" 'yes | kw self-update'
+        ;;
+      'unstable')
+        container_exec "${container}" 'yes | kw self-update --unstable'
+        ;;
+      *)
+        complain "Unsupported update type '${update_type}'"
+        ;;
+    esac
+
+    output=$(container_exec "${container}" 'kw --version')
+
+    assertEquals "(${LINENO}): kw version failed for ${distro}" "$expected_output" "$output"
+  done
+}
+
+function test_self_update_master_helper()
+{
+  self_update_test_helper 'master'
+}
+
+function test_self_update_unstable_helper()
+{
+  self_update_test_helper 'unstable'
+}
+
```

Comment:
One test that I want to see:
1. Manually install kw in the container. Check if everything is all right.
2. Run an update to master (do the same for the unstable) and check if everything looks all right.

Do a similar test, but in this case, intentionally install an old version of the kw (maybe three commits old, you can force it via git) and ensure the update works as expected.